src/ascii_parser.rl

```diff
@@ -0,0 +1,84 @@
+#include "ascii_parser.h"
+
+#define MARK(FPC) parser->mark = FPC - buffer
+#define CAPTURE(LABEL, FPC) do { \
+  parser->LABEL.start = buffer + parser->mark; \
+  parser->LABEL.len = FPC - buffer - parser->mark; \
+} while (0)
+#define CLEAR(LABEL) do { \
+  parser->LABEL.start = NULL; \
+  parser->LABEL.len = 0; \
+} while (0)
+#define EMIT(T) ( ascpp_emit(parser, T) )
+
+/** Machine **/
+%%{
+  machine ascii_protocol_parser;
+
+  action Mark           { MARK(fpc); }
+  action Reset          { CLEAR(name); CLEAR(value); CLEAR(samplerate);}
+  action Name           { CAPTURE(name, fpc); }
+  action Value          { CAPTURE(value, fpc); }
+  action SampleRate     { CAPTURE(samplerate, fpc); }
+  action KeyValue       { EMIT(KEY_VAL); }
+  action Gauge          { EMIT(GAUGE); }
+  action Timer          { EMIT(TIMER); }
+  action Counter        { EMIT(COUNTER); }
+  action Set            { EMIT(SET); }
+
+  action stat_err        { ret=-1; fhold; fgoto line; }
+
+  valstr       = (any - '|')+                       >Mark %Value;
+  name         = (any - ':')+                       >Mark %Name;
+  samplerate   = (any - '\n')+                      >Mark %SampleRate;
+
+  keyvalue     = valstr '|kv'                       %KeyValue;
+  gauge        = valstr '|g'                        %Gauge;
+  timer        = valstr '|' ('h' | 'ms')            %Timer;
+  counter      = valstr ('|c' ('|@' samplerate)?)   %Counter;
+  set          = valstr '|s'                        %Set;
+
+  stat         = (name (':' (keyvalue | gauge | timer | counter | set))+) >Reset $err(stat_err); 
```

Comment:
I've been working on a solution that adds a few things:

- check if the person compiling has ragel, if not, use the release-supplied ragel output
- check if the person compiling has --enable-condensed or not

A few things I haven't 'fixed':

- making sure the release tarball has all the files (ragel input + source output)
- finding a way to merge my changes on top of your changes to then merge it into this repo